vspreview/core/vsenv.py

```diff
@@ -48,40 +49,36 @@ def _monkey_runpy_func(*args: Any, **kwargs: Any) -> Any:
 
     try:
         glob_dict = orig_runpy_run_code(*args, **kwargs)
-    except (AttributeError, ModuleNotFoundError) as e:
-        is_attr = isinstance(e, AttributeError)
+    except ModuleNotFoundError as e:
+        raise DependencyNotFoundError(
+            'vspreview', e.name,
+            f'Could not find the Python package, \'{e.name}\'! '
+            f'Please install it and try again.\n'
+            f'See {guide_url} for installation instructions.'
+        )
+
+    except AttributeError as e:
+        # Only handle VapourSynth-related attribute errors
+        if not hasattr(e, 'obj') or not isinstance(e.obj, core):
```

Comment:
wait what you can do this? does this not produce `TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union`?